api/v1alpha1/authorization_types.go

```diff
@@ -16,50 +17,49 @@ type Authorization struct {
 	// and the second rule denies it, when a request matches both rules, it will be allowed.
 	//
 	// +optional
-	Rules []Rule `json:"rules,omitempty"`
+	Rules []AuthorizationRule `json:"rules,omitempty"`
 
 	// DefaultAction defines the default action to be taken if no rules match.
 	// If not specified, the default action is Deny.
 	// +optional
-	DefaultAction *RuleActionType `json:"defaultAction"`
+	DefaultAction *AuthorizationAction `json:"defaultAction"`
 }
 
-// Rule defines the single authorization rule.
-// +notImplementedHide
-type Rule struct {
+// AuthorizationRule defines a single authorization rule.
+type AuthorizationRule struct {
+	// Name is a user-friendly name for the rule. It's just for display purposes.
```

Comment:
used for logging ?

Reply:
It could be used for logging, but more importanly, an explicit name helps to explain each rule in a `SecurityPolicy`.

Instead of 
```yaml
  authorization:
    defaultAction: Deny
    rules:
    - action: Allow
      principal:
        clientCIDRs:
        - 10.0.2.0/24
    - action: Allow
      principal:
        clientCIDRs:
        - 10.0.1.0/24    
```

We can add a name to each rule so people can know what's the purpose of each rule, like this:

```yaml
  authorization:
    defaultAction: Deny
    rules:
    - name: us-west-region
      action: Allow
      principal:
        clientCIDRs:
        - 10.0.2.0/24
    - name: us-east-region 
      action: Allow
      principal:
        clientCIDRs:
        - 10.0.1.0/24   
```